### PR TITLE
Add helper to reduce header boilerplate

### PR DESCRIPTION
**Pull request by bplommer**
With a random selection of usage examples

### CODE REVIEW

**Comment by bplommer** [comment]
Converted to draft pending bincompat fix

**Review by rossabaker — APPROVED** [review]
I like it.  Inheritance hasn't always been popular around here, so I'll wait for a second approval, but I think it promotes consistency in this case.

**Pull request merged**
